sphinx/builders/linkcheck.py

```diff
@@ -283,13 +285,23 @@ def __init__(self, config: Config,
         self.allowed_redirects = config.linkcheck_allowed_redirects
         self.retries: int = config.linkcheck_retries
         self.rate_limit_timeout = config.linkcheck_rate_limit_timeout
+        self.allow_unauthorized = config.linkcheck_allow_unauthorized
```

Comment:
Since we know that it will be removed, maybe we could make it a private variable?

```suggestion
        self._allow_unauthorized = config.linkcheck_allow_unauthorized
```